src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java

```diff
@@ -0,0 +1,442 @@
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * GCS Plugin related step design.
+ */
+public class GCSConnector implements CdfHelper {
+
+  static {
+    SeleniumHelper.getPropertiesLocators(GCSLocators.class);
+  }
+
+  List<String> propertiesSchemaColumnList = new ArrayList<>();
+  Map<String, String> sourcePropertiesOutputSchema = new HashMap<>();
+
+  @Given("Open Datafusion Project to configure pipeline")
+  public void openDatafusionProjectToConfigurePipeline() throws IOException, InterruptedException {
+    openCdf();
+  }
+
+  @When("Source is GCS bucket")
+  public void sourceIsGCSBucket() throws InterruptedException {
+    CdfStudioActions.selectGCS();
+  }
+
+  @When("Target is GCS bucket")
+  public void targetIsGCSBucket() {
+    CdfStudioActions.sinkGcs();
+  }
+
+  @When("Target is BigQuery")
+  public void targetIsBigQuery() {
+    CdfStudioActions.sinkBigQuery();
+  }
+
+  @Then("Open GCS Properties")
+  public void openGCSProperties() {
+    CdfStudioActions.clickProperties("GCS");
+  }
+
+  @Then("Connect Source as {string} and sink as {string} to establish connection")
+  public void connectSourceAndSinkToEstablishConnection(String source, String sink) throws InterruptedException {
+    CdfStudioActions.connectSourceAndSink(source, sink);
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string}")
+  public void enterTheGCSPropertiesWithGCSBucket(String bucket, String format) throws IOException,
+    InterruptedException {
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.enterReferenceName();
+    CdfGcsActions.getGcsBucket(E2ETestUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat(E2ETestUtils.pluginProp(format));
+    CdfGcsActions.enterSampleSize(E2ETestUtils.pluginProp("gcsSampleSize"));
+    CdfGcsActions.skipHeader();
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with Path Field {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithPathField(String bucket, String format, String pathField)
+    throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterPathField(E2ETestUtils.pluginProp(pathField));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} " +
+    "with Override field {string} and data type {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithOverrideField
+    (String bucket, String format, String overrideField, String dataType) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterOverride(E2ETestUtils.pluginProp(overrideField));
+    CdfGcsActions.clickOverrideDataType(E2ETestUtils.pluginProp(dataType));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with delimiter field {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithDelimiterField(
```

Comment:
Same comment as above have a separate step `Enter GCS property delimiter {string}`

Reply:
Separate steps created for entering properties.